x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportStartTransformAction.java

```diff
@@ -235,7 +240,7 @@ protected void masterOperation(
             );
             transformConfigHolder.set(config);
             ClientHelper.executeWithHeadersAsync(
-                config.getHeaders(),
+                securityHeaders,
```

Comment:
This is a change in behaviour. Currently you don't need to supply secondary headers when starting a transform, but this change would mean that you would have to.

I think the way around it is to change the approach to calling `ValidateTransformAction` completely.

We should call `ValidateTransformAction` using the internal `_xpack` user, i.e. `executeAsyncWithOrigin`. (`ValidateTransformAction` doesn't even have a REST endpoint - from the end user's perspective it doesn't exist. It is entirely an internal implementation detail.) Then, internally, `TransportValidateTransformAction` should execute client calls using `config.getHeaders()`.